src/utils.js

```diff
@@ -478,4 +480,82 @@ export function replaceUnsupportedPDFCharacters(string, font) {
       }
   }
   return String.fromCodePoint(...codePoints);
- }
+}
+
+export const sortByKey = (arr, key) => {
+	return arr.sort((a, b) => {
+		if (a[key] > b[key]) return 1;
+		if (b[key] > a[key]) return -1;
+		return 0;
+	});
+}
+
+/**
+ * Fetches participant data based on the provided query parameters.
+ * 
+ * @async
+ * @function getParticipants
+ * @returns {Promise<Object>} A promise that resolves to the API response containing:
+ *   - code: HTTP status code (200 for success)
+ *   - data: Array of participant objects
+ *   - message: Response message (if applicable)
+ * @throws {Error} If the API request fails
+ */
+export const getParticipants = async () => {
+	
+	const { participantTypeFilter, siteCode, startDateFilter, endDateFilter, cursorHistory, pageNumber, direction } = appState.getState();
+  appState.setState({direction: ``});
+
+	const params = new URLSearchParams();
+	params.append('api', 'getParticipants');
+  params.append('limit', 50);
+	params.append('type', participantTypeFilter || 'all');
+
+	if (siteCode && siteCode !== nameToKeyObj.allResults) {
+		params.append('site', siteCode);
+	}
+
+	if (startDateFilter) {
+		params.append('from', `${startDateFilter}T00:00:00.000Z`);
```

Comment:
Should we consider timezones for start and end dates?

Reply:
The times that are saved into Firestore don't have timezones, so this felt like the proper way to apply the filter.